src/mpyl/artifacts/build_artifacts.py

```diff
@@ -132,6 +134,8 @@ def push(
 
                 artifact_repo.stage(".")
                 artifact_repo.commit(message)
+                if remote_branch_exists:
+                    artifact_repo.pull()  # to prevent issues with parallel runs pushing to the same branch
```

Comment:
If we want to take care of this scenario then I think we need some kind of retry mechanism. Because it could still be that the push of one parallel run happens between the pull and push of the other run